db/migrate/20220331113745_foreman_gce_to_foreman_google_gce.rb

```diff
@@ -1,14 +1,40 @@
+# rubocop:disable Metrics/AbcSize, Metrics/MethodLength
 class ForemanGceToForemanGoogleGce < ActiveRecord::Migration[6.0]
   def up
     User.without_auditing do
-      old_type = 'Foreman::Model::GCE'
-      new_type = 'ForemanGoogle::GCE'
-      compute_resources = ComputeResource.unscoped.where(type: old_type)
+      # First update the type to avoid error:
+      # ActiveRecord::SubclassNotFound: The single-table inheritance
+      # mechanism failed to locate the subclass: 'Foreman::Model::GCE'
+      ComputeResource.unscoped.where(type: original_type).update_all(type: new_type)
 
-      compute_resources.each do |cr|
+      ComputeResource.unscoped.where(type: new_type).each do |cr|
+        unless cr.attrs[:key_path]
+          say("Compute resource [#{cr.name}] is missing path to JSON key file, can't load the data. Please update the resource manually.")
+          next
+        end
         json_data = File.read(cr.attrs[:key_path])
-        cr.update(type: new_type, password: json_data)
+        cr.update(password: json_data)
       end
     end
   end
+
+  def down
+    ComputeResource.unscoped.where(type: new_type).update_all(type: original_type)
+    ComputeResource.unscoped.where(type: original_type).each do |cr|
```

Comment:
> Run rollback
> Checkout to deploy (with GCE in the core code)

Rollback in scenario 2 does not work. `.each` instantiates records and fails to find a class for compute resources of `Foreman::Model::GCE`. It works if I swap the steps.

Also in scenario 2 I had to disable `foreman_google` before the final check otherwise Foreman would refuse to do anything because of pending migrations.